activerecord/lib/active_record/inheritance.rb

```diff
@@ -49,8 +49,18 @@ def new(*args, &block)
         if abstract_class? || self == Base
           raise NotImplementedError, "#{self} is an abstract class and cannot be instantiated."
         end
+        
```

Comment:
Instead of this custom implementation should not we wrap the whole code inside the `scoping` block like we do in relation.rb's `new` implementaion?